src/PublicApiGenerator/CodeTypeReferenceBuilder.cs

```diff
@@ -0,0 +1,142 @@
+﻿using Mono.Cecil;
+using System;
+using System.CodeDom;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace PublicApiGenerator
+{
+    internal static class CodeTypeReferenceBuilder
+    {
+        private const int MAX_COUNT = 1000;
+
+        internal static CodeTypeReference CreateCodeTypeReference(TypeReference type, ICustomAttributeProvider attributeProvider)
```

Comment:
Would it make sense to change this to 

```
CreateCodeTypeReference(TypeReference type, ICustomAttributeProvider attributeProvider = null)
```

to avoid the `null` passing above?

Reply:
Also should these be extension methods on `TypeReference`?

Reply:
It may be an extension method. Ok.

Reply:
Done